src/util/fs.ts

```diff
@@ -0,0 +1,263 @@
+// Like node fs/promises, only we maintain a budget of open files to prevent
```

Comment:
```suggestion
/**
 * @license
 * Copyright 2023 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

// Like node fs/promises, only we maintain a budget of open files to prevent
```